### ISSUE TITLE
Windows: Path to client.crt is calculated incorrectly

### ISSUE THREAD

**Issue by guibirow** [kubernetes/minikube#4053]
While trying to run ```minikube start``` using minikube version: **v1.0.0** on Windows 10, with following config:

```
{
    "WantReportError": true,
    "WantReportErrorPrompt": false,
    "dashboard": true,
    "hyperv-virtual-switch": "Default Switch",
    "profile": "minikube",
    "vm-driver": "hyperv"
}
```
I get the following error:
```
C:\Users\Diego.Mendes>minikube start
o   minikube v1.0.0 on windows (amd64)
$   Downloading Kubernetes v1.14.0 images in the background ...
>   Creating hyperv VM (CPUs=2, Memory=2048MB, Disk=20000MB) ...
@   Downloading Minikube ISO ...
 142.88 MB / 142.88 MB [============================================] 100.00% 0s
-   "minikube" IP address is 172.17.57.124
-   Configuring Docker as the container runtime ...
-   Version of container runtime is 18.06.2-ce
:   Waiting for image downloads to complete ...
-   Preparing Kubernetes environment ...
@   Downloading kubelet v1.14.0
@   Downloading kubeadm v1.14.0
-   Pulling images required by Kubernetes v1.14.0 ...
-   Launching Kubernetes v1.14.0 using kubeadm ...
:   Waiting for pods:
!   Error starting cluster: wait: k8s client: Error creating kubeConfig: invalid configuration: [unable to read client-cert C:\Users\Diego.Mendes\.kube\Users\Diego.Mendes\.minikube\client.crt for minikube due to open C:\Users\Diego.Mendes\.kube\Users\Diego.Mendes\.minikube\client.crt: The system cannot find the path specified., unable to read client-key C:\Users\Diego.Mendes\.kube\Users\Diego.Mendes\.minikube\client.key for minikube due to open C:\Users\Diego.Mendes\.kube\Users\Diego.Mendes\.minikube\client.key: The system cannot find the path specified., unable to read certificate-authority C:\Users\Diego.Mendes\.kube\Users\Diego.Mendes\.minikube\ca.crt for minikube due to open C:\Users\Diego.Mendes\.kube\Users\Diego.Mendes\.minikube\ca.crt: The system cannot find the path specified.]

*   Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
-   https://github.com/kubernetes/minikube/issues/new
```

Looking into details, I noticed the PATH to the certificates are invalid, it tries:
```C:\Users\Diego.Mendes\.kube\Users\Diego.Mendes\.minikube\client.crt```

Where should should be:
```C:\Users\Diego.Mendes\.minikube\client.crt```

Minikube is trying to find the certificates adding an extra ```.kube\Users\Diego.Mendes``` that does not exist. I couldn't find the proper way to set the path, so I decided to copy the certificates to the path and it works around the problem.

PS: I had to delete and recreate the cluster to make it work properly.

**Comment by tstromberg** [comment]
Interesting. It sounds like a windows-specific path bug, but I really have no idea why it happened.

**Comment by medyagh** [comment]
@guibirow do you still have this issue with latest minikube ?

**Comment by ltomes** [comment]
@medyagh I currently have this issue with minikube 1.3.1

**Comment by medyagh** [comment]
do you mind sharing the ENV variables of 
MINIKUBE_HOME
and 
KUBECONFIG

**Comment by medyagh** [comment]
> @medyagh I currently have this issue with minikube 1.3.1
> 
> Edit: My issue was related to the virtual switch being used, pointing it at the same switch being used by WSL seems to work for my needs:
> `minikube start --vm-driver=hyperv --hyperv-virtual-switch=WSL`

so you dont have this issue ?

**Comment by medyagh** [comment]
> @medyagh I currently have this issue with minikube 1.3.1
> 
> Edit: My issue was related to the virtual switch being used, pointing it at the same switch being used by WSL seems to work for my needs:
> `minikube start --vm-driver=hyperv --hyperv-virtual-switch=WSL`

do you mind sharing the path of your `client.crt`

**Comment by ltomes** [comment]
I think my issue is different / I can't communicate properly with kubectl either. 
```
 kubectl get namespace default
The connection to the server kubernetes.docker.internal:6443 was refused - did you specify the right host or port?
```
I'm pretty sure I just have a configuration issue with my virtual switch. Sorry for muddying up this ticket!

**Comment by medyagh** [comment]
> I think my issue is different / I can't communicate properly with kubectl either.
> 
> ```
>  kubectl get namespace default
> The connection to the server kubernetes.docker.internal:6443 was refused - did you specify the right host or port?
> ```
> 
> I'm pretty sure I just have a configuration issue with my virtual switch. Sorry for muddying up this ticket!

If you dont mind I am still interested to see the path for the client.crt

**Comment by guibirow** [comment]
Hi, sorry, I haven't updated my minikube since raised this issue.

I will try it on Friday, I don't have my laptop with me now.

**Comment by ltomes** [comment]
> If you dont mind I am still interested to see the path for the client.crt
@medyagh 
`.kube/config` lists the following paths:
```
users:
...
- name: minikube
  user:
    client-certificate: C:\Users\<user>\.minikube\client.crt
    client-key: C:\Users\<user>\.minikube\client.key
```

**Comment by medyagh** [comment]
Thanks

**Comment by medyagh** [comment]
@guibirow unfortunately I can not reproduce this issue, I am gonna close this, please feel free to re-open if you still have this issue in the latest version of minikube

**Comment by guibirow** [comment]
So, I just had time to test it.

Neither env vars are set in my environment

The `.kube/config` file set the client certs like this:
```
- cluster:
    certificate-authority: \Users\Diego.Mendes\.minikube\ca.crt
    server: https://192.168.52.14:8443
  name: minikube

- cluster:
    certificate-authority: \Users\Diego.Mendes\.minikube\ca.crt
    server: https://192.168.52.2:8443
  name: minikube2
```
As you can see, it does not use the drive letter as root, it is a valid root path in windows like Linux `/users/name`, probably the script expect the drive letter, and because it does not find one it build the wrong path. it should assume starting with `\` to be root and not prepend anything.

These two env vars are set, might help diagnose:
```
HOMEDRIVE=C:
HOMEPATH=\Users\Diego.Mendes
```

Using the minikube v1.3.1 has the same issues:
```
C:\Users\Diego.Mendes>minikube start -p minikube2
* [minikube2] minikube v1.3.1 on Microsoft Windows 10 Pro 10.0.18362 Build 18362
* Downloading VM boot image ...
minikube-v1.3.0.iso.sha256: 65 B / 65 B [--------------------] 100.00% ? p/s 0s
minikube-v1.3.0.iso: 131.07 MiB / 131.07 MiB [-------] 100.00% 4.60 MiB p/s 28s
* Creating hyperv VM (CPUs=2, Memory=2000MB, Disk=20000MB) ...
* Preparing Kubernetes v1.15.2 on Docker 18.09.8 ...
* Downloading kubelet v1.15.2
* Downloading kubeadm v1.15.2
* Pulling images ...
* Launching Kubernetes ...
*
X Error starting cluster: timed out waiting to elevate kube-system RBAC privileges: getting clientset: kubeConfig: invalid configuration: [unable to read client-cert C:\Users\Diego.Mendes\.kube\Users\Diego.Mendes\.minikube\client.crt for minikube2 due to open C:\Users\Diego.Mendes\.kube\Users\Diego.Mendes\.minikube\client.crt: The system cannot find the path specified., unable to read client-key C:\Users\Diego.Mendes\.kube\Users\Diego.Mendes\.minikube\client.key for minikube2 due to open C:\Users\Diego.Mendes\.kube\Users\Diego.Mendes\.minikube\client.key: The system cannot find the path specified., unable to read certificate-authority C:\Users\Diego.Mendes\.kube\Users\Diego.Mendes\.minikube\ca.crt for minikube2 due to open C:\Users\Diego.Mendes\.kube\Users\Diego.Mendes\.minikube\ca.crt: The system cannot find the path specified.]
*
* Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
  - https://github.com/kubernetes/minikube/issues/new/choose
* Problems detected in kube-addon-manager:
  - error: unable to recognize "STDIN": Get https://localhost:8443/api?timeout=32s: dial tcp 127.0.0.1:8443: connect: connection refused
```

I've also tried cleaning the machine from any minikube configuration, delete the `.minikube` folder and the problem still persists.


I'm using Microsoft Windows 10 Pro 10.0.18362.239 Build 18362 [Release 1903]

**Comment by guibirow** [comment]
I can confirm that setting the `MINIKUBE_HOME` with the following command will fix the problem:
```set MINIKUBE_HOME=%HOMEDRIVE%%HOMEPATH%\.minikube```
And running the following command:
```
C:\Users\Diego.Mendes>
minikube start -p minikube2 --vm-driver=hyperv --hyperv-virtual-switch="Default Switch"

* [minikube2] minikube v1.3.1 on Microsoft Windows 10 Pro 10.0.18362 Build 18362
* Creating hyperv VM (CPUs=2, Memory=2000MB, Disk=20000MB) ...
* Preparing Kubernetes v1.15.2 on Docker 18.09.8 ...
* Pulling images ...
* Launching Kubernetes ...
* Waiting for: apiserver proxy etcd scheduler controller dns
* Done! kubectl is now configured to use "minikube2"
```
Will create a new cluster properly

I think this should be done by **minikube** and I shouldn't need to do that myself

I can't reopen the bug, so I would be glad if reopen it, please!

**Comment by medyagh** [comment]
@guibirow thanks for updating the issue. I couldn't produce this issue on our windows machine. And others don't have this issue.

Do you mind sharing more info about your windows version?

Is it possible to that the MINIKUBE_HOME was set before to another path ?

**Comment by guibirow** [comment]
Windows 10 Pro 10.0.18362.239 Build 18362 [Release 1903]

There was no value for `MINIKUBE_HOME` in the env vars
I will try again tonight, 
I will remove the environment to see if the problem return

**Comment by guibirow** [comment]
Looks like I found the cause of the issue

I have removed the previous created clusters, deleted the env var `MINIKUBE_HOME` and tried again, the error returned.

After a lot of back and forth, I found out that the conflict is in the `home` env var. The current value is 
`home=\Users\Diego.Mendes`, this will cause minikube to prepend the `C:\Users\Diego.Mendes\.kube` at the beginning of the path causing the bug.

When I set my env var to `home=C:\Users\Diego.Mendes\.kube` with the drive letter, it works fine.

it shouldn't be required, because `home=\Users\Diego.Mendes` is a valid path.

I think whatever is the logic that handles this, should use either `HOMEPATH` or `home` the way it is without modification.

I don't have go experience, otherwise I could investigate the code.

I hope this help you reproduce the issue.
